zeek

```diff
@@ -89,15 +63,58 @@ start)
 		exit 0
 	fi
 
+	# create bare minimum directories required for running Zeek
+	if [ ! -d "$HOST_ZEEK/logs" ] || [ ! -d "$HOST_ZEEK/spool" ] || [ ! -d "$HOST_ZEEK/etc" ]; then
+		sudo mkdir -p "$HOST_ZEEK/logs" "$HOST_ZEEK/spool" "$HOST_ZEEK/etc"
+	fi
+
+	# create the node.cfg file required for running Zeek
+	if [ ! -s "$HOST_ZEEK/etc/node.cfg" ]; then
+		echo "Could not find $HOST_ZEEK/etc/node.cfg. Generating one now." >&2
+
+		$SUDO docker run --rm -it --network host \
+		--mount source="$HOST_ZEEK/etc/",destination=/zeek/,type=bind \
+		"$IMAGE_NAME" \
+		zeekcfg -o /zeek/node.cfg --type afpacket
+	fi
+
+	# create the volumes required for peristing user-installed zkg packages
+	$SUDO docker volume create zeek-zkg-script >/dev/null
+	$SUDO docker volume create zeek-zkg-plugin >/dev/null
+	$SUDO docker volume create zeek-zkg-state >/dev/null
+
+	docker_cmd=("docker" "run" "--detach")  # start container in the background
+	docker_cmd+=("--name" "$CONTAINER_NAME") # provide a predictable name
+	docker_cmd+=("--restart" "$RESTART_POLICY")
+	docker_cmd+=("--cap-add" "net_raw" "--cap-add" "net_admin") # allow Zeek to listen to raw packets
+	docker_cmd+=("--network" "host") # allow Zeek to monitor host network interfaces
+	# allow packages installed via zkg to persist across restarts
+	docker_cmd+=("--mount" "source=zeek-zkg-script,destination=/usr/local/zeek/share/zeek/site/packages/,type=volume")
+	docker_cmd+=("--mount" "source=zeek-zkg-plugin,destination=/usr/local/zeek/lib/zeek/plugins/packages/,type=volume")
+	docker_cmd+=("--mount" "source=zeek-zkg-state,destination=/root/.zkg,type=volume")
+	# mirror the host timezone settings to the container
+	docker_cmd+=("--mount" "source=/etc/localtime,destination=/etc/localtime,type=bind,readonly")
+	# persist and allow accessing the logs from the host
+	docker_cmd+=("--mount" "source=$HOST_ZEEK/logs,destination=/usr/local/zeek/logs/,type=bind")
+	docker_cmd+=("--mount" "source=$HOST_ZEEK/spool,destination=/usr/local/zeek/spool/,type=bind")
+	# allow users to provide arbitrary custom config files and scripts
+	# mount all zeekctl config files
+	while IFS=  read -r -d $'\0' CONFIG; do
+    	docker_cmd+=("--mount" "source=$CONFIG,destination=/usr/local/zeek/${CONFIG#"$HOST_ZEEK"},type=bind")
+	done < <(find "$HOST_ZEEK/etc/" -type f -print0 2>/dev/null)
+	# mount all zeek scripts
+	while IFS=  read -r -d $'\0' SCRIPT; do
+    	docker_cmd+=("--mount" "source=$SCRIPT,destination=/usr/local/zeek/${SCRIPT#"$HOST_ZEEK"},type=bind")
+	done < <(find "$HOST_ZEEK/share/" -type f -print0 2>/dev/null)
```

Comment:
These two loops are a bit clever (read: arcane and unreadable). But they allow users to add in custom scripts and files without 1) us having to know what they are ahead of time and 2) mounting a whole directory and hiding additional files in the Zeek install. Any files put in `zeek/etc` or `zeek/share` will get this treatment and automatically override the corresponding file inside.